timesketch/lib/analyzers/sequence_sessionizer.py

```diff
@@ -0,0 +1,152 @@
+        should be matched.
+        recording: Shows if currently are finded and stored events that match
+        the specified sequence of events.
+        return_fields: List of name of event attributes, should be specified
+        in the inheriting sessionizers. It's mandatory to contains 'timestamp'.
+        session_num: Counter for the number of sessions.
+    """
+    event_seq_name = None
+    event_seq = None
+    event_storage = []
+    num_event_to_find = 0
+    recording = False
+    return_fields = ['timestamp']
+    session_num = 0
+
+    def run(self):
+        """Entry point for the analyzer. Allocates each event between the
+        first event of the event_seq and the last event of the event_seq
+        an event_seq_name attribute and a session_num.
+
+        Returns:
+            String containing the name of the event sequence and the
+            number of sessions created.
+        """
+        if self.event_seq_name is None:
+            raise RuntimeError('No event_seq_name provided.')
+        if self.event_seq is None or self.event_seq == []:
+            raise RuntimeError('No event_seq provided.')
+        if 'timestamp' not in self.return_fields:
+            raise RuntimeError('No \'timestamp\' in return_fields.')
+        if len(self.return_fields) <= 1:
+            raise RuntimeError('No return_fields provided.')
+
+        # event_stream returns an ordered generator of events (by time)
+        # therefore no further sorting is needed.
+        events = self.event_stream(query_string=self.query,
+                                   return_fields=self.return_fields)
+
+        last_timestamp = 0
+
+        try:
+            first_event = next(events)
+            last_timestamp = first_event.source.get('timestamp')
+            self.process_event(first_event)
+
+            for event in events:
+                curr_timestamp = event.source.get('timestamp')
+                if curr_timestamp - last_timestamp > self.max_time_diff_micros:
+                    self.flush_events(drop=True)
+
+                self.process_event(event)
+                last_timestamp = curr_timestamp
+        except StopIteration:
+            pass
+
+        self.query = self.get_query_string()
+
+        self.sketch.add_view('Session view', self.NAME, query_string=self.query)
+
+        return ('Sessionizing completed, number of {0} session created:'
+                ' {1:d}'.format(self.event_seq_name, self.session_num))
+
+    def annotateEvent(self, event, session_num):
+        """Add an event_seq_name attribute with a session_num to event.
+
+        Args:
+            event: Event to annotate.
+            session_num: Session number for the event.
+        """
+        event.add_attributes({self.event_seq_name: session_num})
+        event.commit()
+
+    def process_event(self, event):
+        """Process event depending on if the event is significant for the
```

Comment:
Same here, try to have a docstring with a brief one-line description, and insert details in a paragraph underneath.